src/ResourceManager/CognitiveServices/Commands.Management.CognitiveServices/CognitiveServicesAccount/GetAzureCognitiveServicesAccountSkus.cs

```diff
@@ -15,45 +15,90 @@
 using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
 using Microsoft.Azure.Management.CognitiveServices;
 using Microsoft.Azure.Management.CognitiveServices.Models;
+using Microsoft.Azure.Commands.Management.CognitiveServices.Models;
+using System;
+using System.Linq;
 using System.Management.Automation;
-using CognitiveServicesModels = Microsoft.Azure.Commands.Management.CognitiveServices.Models;
 
 namespace Microsoft.Azure.Commands.Management.CognitiveServices
 {
     /// <summary>
     /// Get Available Skus for Cognitive Services Account
     /// </summary>
-    [Cmdlet("Get", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "CognitiveServicesAccountSkus"), OutputType(typeof(CognitiveServicesModels.PSCognitiveServicesSkus))]
+    [Cmdlet("Get", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "CognitiveServicesAccountSkus", DefaultParameterSetName = GetSkusWithAccountParamSetName),
+        OutputType(typeof(PSCognitiveServicesSkus), typeof(ResourceSku))]
     public class GetAzureCognitiveServicesAccountSkusCommand : CognitiveServicesAccountBaseCmdlet
     {
+        protected const string GetSkusWithAccountParamSetName = "GetSkusWithAccount";
+        protected const string GetSkusWithFilterParamSetName = "GetSkusWithFilter";
+
+        [Parameter(
+            ParameterSetName = GetSkusWithFilterParamSetName,
+            Position = 0,
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Cognitive Services Account Type.")]
+        [Alias(CognitiveServicesAccountTypeAlias, AccountTypeAlias, KindAlias)]
+        public string Type { get; set; }
+
+        [Parameter(
+            ParameterSetName = GetSkusWithFilterParamSetName,
+            Position = 1,
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Cognitive Services Account Location.")]
+        [LocationCompleter("Microsoft.CognitiveServices/accounts")]
+        public string Location { get; set; }
+
         [Parameter(
+            ParameterSetName = GetSkusWithAccountParamSetName,
             Position = 0,
             Mandatory = true,
             ValueFromPipelineByPropertyName = true,
             HelpMessage = "Resource Group Name.")]
         [ResourceGroupCompleter()]
-        [ValidateNotNullOrEmpty]
```

Comment:
@yangyuan any reason we're removing the `ValidateNotNullOrEmpty` attribute from the existing cmdlets?

Reply:
I removed them as I changed them to NonMandatory parameter.
Added them back as we are using Parameter Set now.